flow/project.py

```diff
@@ -1388,6 +1391,131 @@ def copy_from(cls, *other_funcs):
+                The operation function.
+            """
+
+            _parent_class = parent_class
+
+            def __init__(self, func, name=None):
+                """We somewhat abuse the __init__ function to act as a decorator.
+
+                We return the original function here, and only use classmethods
+                of this class.
+                """
+                if isinstance(func, str):
+                    return lambda op: self(op, name=func)
+
+                if func in chain(
+                    *self._parent_class._OPERATION_PRECONDITIONS.values(),
+                    *self._parent_class._OPERATION_POSTCONDITIONS.values(),
+                ):
+                    raise ValueError(
+                        "A condition function cannot be used as an operation."
+                    )
+
+                if name is None:
+                    name = func.__name__
+
+                for (
+                    registered_name,
+                    registered_func,
+                ) in self._parent_class._OPERATION_FUNCTIONS:
+                    if name == registered_name:
+                        raise ValueError(
+                            f"An operation with name '{name}' is already registered."
+                        )
+                    if func is registered_func:
+                        raise ValueError(
+                            "An operation with this function is already registered."
+                        )
+                if name in self._parent_class._GROUP_NAMES:
+                    raise ValueError(
+                        f"A group with name '{name}' is already registered."
+                    )
+
+                signature = inspect.signature(func)
+                for i, parameter_value in enumerate(signature.parameters.values()):
+                    if i > 0 and parameter_value.default is inspect.Parameter.empty:
+                        raise ValueError(
+                            "Only the first argument in an operation argument may not have "
+                            f"a default value! ({name})"
+                        )
+                if not getattr(func, "_flow_aggregate", False):
+                    func._flow_aggregate = _aggregator.groupsof(1)
+
+                # Append the name and function to the class registry
+                self._parent_class._OPERATION_FUNCTIONS.append((name, func))
+                self._parent_class._GROUPS.append(
+                    FlowGroupEntry(name=name, options="")
+                    # TODO: Enable aggregator argument.
+                    # FlowGroupEntry(name=name, options="", aggregator=func._flow_aggregate)
+                )
+                if hasattr(func, "_flow_groups"):
+                    func._flow_groups.append(name)
+                else:
+                    func._flow_groups = [name]
+                return func
+
+            @classmethod
+            def with_directives(cls, directives, name=None):
```

Comment:
I would prefer 
```suggestion
            def with_directives(cls, name, **directives):
```
as originally proposed in #309.

Reply:
We mention keeping the same API as in `FlowGroup.with_directives` which uses a dictionary.